src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -402,6 +396,16 @@ ClientRequest::process_op(
     co_return;
   }
 
+  // obc is obtained, check if we can serve a replicated read on the resolved object
+  if (!pg->is_primary() &&
+      !pg->get_peering_state().can_serve_replica_read(obc_manager.get_obc()->get_oid())) {
+      DEBUGDPP("{}.{}: unstable write on replica, bouncing to primary",
+               *pg, *this, this_instance_id);
+      pg->get_perf_logger().inc(l_osd_replica_read_redirect_conflict);
+      co_await reply_op_error(pg, -EAGAIN);
+      co_return;
+  }
```

Comment:
Same idea -- let's instead change can_serve_replica_read to simply check for *any* head or clone of the object and leave it where it was.

Reply:
IIUC we have 2 options here:
1) Move `can_serve_replica_read` to check the **resolved** object.
2) Check _any head or clone of the object_

For 2, we would need to iterate over each possible head/clone combination (similar to the previous version of this PR). Am I right?
IMO, checking only the resolved object seems more elegant, unless I'm missing a better way to implement the 2nd option above.

---

Note: Classic seems to call `can_serve_replica_read` for the unresolved object. I'll address this in a followup PR.

Reply:
One downside with moving `can_serve_replica_read` is that we check for writes much later in the flow, meaning we could have bounce this read earlier.

Reply:
Looking again, checking writes on the clone actually checks writes on the head only. This actually makes sense as clone objects **can't** have any independent log entries. In any log entry of a clone (which is a creation of it or deletion/update of it when trimming) there must be also a separate log entry of the head object.

I kept the can_resolve in the original place of it - see last commit for clarification notes. What do you think?